homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  var ListManager = function(){
+  	this._listNoteContent = [];
+	this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
+	start(){
+		this.fetchList().then((function(data){
+			this.updateList(data);
+			this.drawList();
+			this.preloadFirstNote();
+		}).bind(this)).catch((function(){
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+		}).bind(this));
+		window.addEventListener('click',(function(event){
+			this.onNoteOpen(event);
```

Comment:
It's a convention to prefix `on` with a method handling the corresponding event. So it should be renamed as `onClick` or `onNoteClicked` actually. This is a hidden issue about convention and expressiveness.